pkg/types/openstack/machinepool.go

```diff
@@ -77,8 +85,14 @@ type RootVolume struct {
 	// Required
 	Size int `json:"size"`
 	// Type defines the type of the volume.
-	// Required
-	Type string `json:"type"`
+	// Deprecated: Use Types instead.
+	// +optional
+	DeprecatedType string `json:"type,omitempty"`
+
+	// Types is the list of the volume types of the root volumes.
+	// This is mutually exclusive with Type.
+	// +required
```

Comment:
LGTM. This is my own ignorance of this detail of how golang unmarshaling works. Thanks!

Reply:
The job failed later for another reason but the install-config was well converted:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_installer/7300/pull-ci-openshift-installer-master-e2e-openstack-kuryr/1681667818954690560/artifacts/e2e-openstack-kuryr/openstack-conf-installconfig/artifacts/install-config.yaml

Reply:
I'm now seeing this error in another PR:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41450/rehearse-41450-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-kuryr/1681836643008385024/artifacts/e2e-openstack-kuryr/ipi-install-install/build-log.txt

But I suspect it's because the step registry doesn't have the latest installer, whereas the previous step registry which generated the install-config and converted it, has the latest, since we can see the new parameter:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41450/rehearse-41450-periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.14-e2e-openstack-kuryr/1681836643008385024/artifacts/e2e-openstack-kuryr/openstack-conf-installconfig/artifacts/install-config.yaml

Reply:
I'll keep an eye on that, but again both CI and manual tests confirmed that it's supposed to work...

Reply:
Testing again with `pull-ci-openshift-installer-master-e2e-openstack-kuryr` via https://github.com/openshift/release/pull/41450 which might gives a more recent build for the installer.